src/types.ts

```diff
@@ -125,3 +133,32 @@ export interface PluginConfigVMReponse {
     }
     tasks: Record<string, PluginTask>
 }
+
+// received via Kafka
+interface EventMessage {
+    distinct_id: string
+    ip: string
+    site_url: string
+    team_id: number
+}
+
+export interface RawEventMessage extends EventMessage {
+    data: string
+    now: string
+    sent_at: string // may be an empty string
+}
+
+export interface ParsedEventMessage extends EventMessage {
+    data: EventData
+    now: DateTime
+    sent_at: DateTime | null
+}
+
+export interface EventData extends PluginEvent {
+    properties?: Properties
+    timestamp?: string
+    $set?: Properties
+    offset?: number
+}
+
+export type Properties = Record<string, any>
```

Comment:
Could it make sense to move some of this (or all of this) inside the `PluginEvent` type inside the `posthog-plugins` package directly?

Reply:
I'll look into it, but the way the plugin server handles this sort of type doesn't seem entirely consistent with the rest of PostHog (which isn't really consistent with itself at all, given all the legacy stuff and cruft…), so not sure.

Reply:
It seems to make sense actually. Done.